packages/subcommands/src/lib/Subcommand.ts

```diff
@@ -15,10 +16,94 @@ import {
 
 export class Subcommand<PreParseReturn extends Args = Args, O extends Subcommand.Options = Subcommand.Options> extends Command<PreParseReturn, O> {
 	public parsedSubcommandMappings: SubcommandMappingArray;
+	public caseInsensitiveSubcommands = false;
 
 	public constructor(context: PieceContext, options: O) {
 		super(context, options);
 		this.parsedSubcommandMappings = options.subcommands ?? [];
+
+		const clientOptions = this.container.client.options;
+
+		if (clientOptions.caseInsensitiveCommands) {
+			this.caseInsensitiveSubcommands = true;
+
+			// Because slash commands must be lowercase anyway, we can transform all to lowercase.
+			for (const cmd of this.parsedSubcommandMappings) {
+				cmd.name = cmd.name.toLowerCase();
+				if (cmd.type === 'group') {
+					for (const groupCommand of cmd.entries) {
+						groupCommand.name = groupCommand.name.toLowerCase();
+					}
+				}
+			}
+		}
+
+		if (options.generateDashLessAliases) {
+			for (const mapping of this.parsedSubcommandMappings) {
+				if (!Reflect.has(mapping, 'messageRun')) {
+					continue;
+				}
+
+				const dashLessMappings: SubcommandMappingArray = [];
+
+				if (mapping.type === 'group') {
+					// Keep track of whether we have changed entries or not, so we don't need to do expensive object comparison later.
+					let hasChangedEntries = false;
+
+					// Deeply clone the entire mapping to avoid mutating the original.
+					const clonedMapping = deepClone(mapping);
+
+					// Loop through the group's subcommands and add a dash-less alias for each, if applicable.
+					for (const groupCommand of mapping.entries) {
+						if (groupCommand.name.includes('-')) {
+							// If we are inside this if statement then we flip `hasChangedEntries` to true to be able to read it later.
+							hasChangedEntries = true;
+
+							clonedMapping.entries.push({
+								...groupCommand,
+								name: groupCommand.name.replaceAll('-', '')
+							});
+						}
+					}
+
+					/**
+					 * If any of the entries in the group had a dash then `hasChangedEntries` was flipped to true
+					 * and so we want to register a mapping with the base name and all dash-less entries
+					 * in the group.
+					 */
+					if (hasChangedEntries) {
+						dashLessMappings.push({
+							...mapping,
+							entries: clonedMapping.entries
+						});
+					}
+
+					/**
+					 * If the group name itself has a dash then register a dash-less copy of it.
+					 * If the entries in the group didn't have any dashes then `clonedMapping.entries` is still
+					 * the same original array, so it can be safely overwritten.
+					 *
+					 * If the entries in the group did have any dashes then `clonedMapping.entries` has
+					 * all entries both with and without dashes.
+					 */
+					if (mapping.name.includes('-')) {
```

Comment:
This is still wrong! If the group mapping has dashes in its name, a whole different mapping should be created for it, not altering the existing mapping!